pkg/kubelet/apis/config/v1alpha1/defaults.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeletconfigv1alpha1 "k8s.io/kubelet/config/v1alpha1"
+	credentialproviderv1alpha1 "k8s.io/kubernetes/pkg/credentialprovider/apis/credentials/v1alpha1"
+)
+
+func addDefaultingFuncs(scheme *runtime.Scheme) error {
+	return RegisterDefaults(scheme)
+}
+
+func SetDefaults_CredentialProviderConfig(obj *kubeletconfigv1alpha1.CredentialProviderConfig) {
+	for i := range obj.Providers {
+		if obj.Providers[i].APIVersion == "" {
+			obj.Providers[i].APIVersion = credentialproviderv1alpha1.SchemeGroupVersion.String()
```

Comment:
> If we intend to stop defaulting this and make it required in v1beta1, add a comment here so we don't blindly copy this to v1beta1

Can we just make it required now?

Reply:
definitely. I probably would. It's slightly silly (because there's only a single valid value currently) but we know we'll expand it to beta and v1 in the future

Reply:
makes sense, will update